src/shims/posix/fs.rs

```diff
@@ -223,7 +223,7 @@ impl<'tcx> FileHandler {
             self.handles.last_key_value().map(|(fd, _)| fd.checked_add(1).unwrap()).unwrap_or(min_fd)
         });
 
-        self.handles.insert(new_fd, file_handle).unwrap_none();
+        assert_eq!(self.handles.insert(new_fd, file_handle), None);
```

Comment:
I think these need to use `is_none()`, there is no `==` op for `Box<dyn ...>`

Reply:
Oh oops. I thought I tested the code. Forgot to run `check` apparently.